File: code/generate_strategies/IR(pseudocode)/java/java/function_tenacity_retry_if_exception_type.java
package com.example;

import dev.failsafe.Failsafe;
import dev.failsafe.RetryPolicy;
import dev.failsafe.function.CheckedSupplier;
import java.time.Duration;

public class RetryFunction {
    public int might_fail_function(int value, boolean shouldRaise) {
        
        class ValueError extends RuntimeException {
            public ValueError(String message) {
                super(message);
            }
        }

        
        RetryPolicy<Object> retryPolicy = RetryPolicy.builder()
                .handle(ValueError.class)
                .withMaxAttempts(3)
                .withDelay(Duration.ofSeconds(1))
                .build();

        
        CheckedSupplier<Integer> supplier = () -> {
            System.out.println("The current value is: " + value);
            try {
                Thread.sleep(100); 
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
            if (shouldRaise) {
                throw new ValueError("Value is too small");
            }
            return value;
        };

        
        return Failsafe.with(retryPolicy).get(supplier);
    }
}

File: manual/data/python/src/function_tenacity_retry_if_exception_type.py
import time
from tenacity import retry, stop_after_attempt, wait_fixed, retry_if_exception_type

class RetryFunction:
    @retry(stop=stop_after_attempt(3), wait=wait_fixed(1), retry=retry_if_exception_type(ValueError))
    def might_fail_function(self, value: int, should_raise: bool = False) -> int:
        print(f"The current value is: {value}")
        time.sleep(0.1)  
        if should_raise:
            raise ValueError("Value is too small")
        return value
